### PR TITLE
L.BatchNormalization: miscellaneous fixes

### PR DESCRIPTION
**Pull request by toslunar**
This PR fixes some issues I wasn't aware of when I reviewed #4266.
- Move the document of `axis` to the correct position
- Add `axis` test to links_test

### CODE REVIEW

File: tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py [hunk 1]
```diff
@@ -23,18 +23,45 @@ def _batch_normalization(expander, gamma, beta, x, mean, var, eps, test):
...
-    'dtype': [numpy.float16, numpy.float32, numpy.float64],
-})))
+@testing.parameterize(*(testing.product_dict(
+    testing.product({
+        'test': [True, False],
+        'dtype': [numpy.float16, numpy.float32, numpy.float64],
+    }),
+    testing.product({
+        'ndim': [0, 1, 2, 3],
+        'axis': [None],
```
Comment (okuta) [inline tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py]: Is it correct?

**Comment by okuta** [comment]
jenkins, test this please.

File: tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py (hunk 1 above)
Comment (okuta) [reply tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py]: Yes, it is prefer. Another way is to divide `BatchNormalizationTest` class.

**Comment by toslunar** [comment]
Fixed!

**Comment by okuta** [comment]
LGTM!

**Pull request merged**